### PR TITLE
Fix no_std builds attempt 2

### PR DESCRIPTION
**Pull request by trevor-crypto**
- Turn off default features for core in gen and genmult
- Use resolver="2"

Relates to #71

### CODE REVIEW

**Comment by athei** [comment]
The [CI should check](https://github.com/paritytech/libsecp256k1/blob/c0def35986fcf98eb0cd2ec1d62fb764cbca2ea3/.github/workflows/rust.yml#L339-L344) whether `no_std` builds are working. Any idea why this wasn't caught there?

File: Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ authors = ["Wei Tang <hi@that.world>"]
 repository = "https://github.com/paritytech/libsecp256k1"
 keywords = ["crypto", "ECDSA", "secp256k1", "bitcoin", "no_std"]
 edition = "2018"
+resolver = "2"
```
Comment (sorpaas) [inline Cargo.toml:10]: This should be enough. Please revert the rest of the changes.

Comment (trevor-crypto) [reply Cargo.toml:10]: Done.

**Comment by trevor-crypto** [comment]
> The [CI should check](https://github.com/paritytech/libsecp256k1/blob/c0def35986fcf98eb0cd2ec1d62fb764cbca2ea3/.github/workflows/rust.yml#L339-L344) whether `no_std` builds are working. Any idea why this wasn't caught there?

@athei I don't think that actually checks against a no-std target. You need to add a target like `thumbv6m-none-eabi`

**Comment by athei** [comment]
The only difference I see is that your command line is cross compiling where the CI does not. Both use `--no-default-features`. It that correct?

**Comment by trevor-crypto** [comment]
@athei they both use `--no-default-features` but that doesn't guarantee that the library will build successfully for a target without `std`. You can also use something like https://github.com/hobofan/cargo-nono to do a check

**Pull request merged**